rosidl_typesupport_c/Doxyfile

```diff
@@ -0,0 +1,29 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rosidl_typesupport_c"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "ROSIDL C Typesupport"
+
+# Use these lines to include the generated logging.hpp (update install path if needed)
+INPUT                  = README.md ./include
+USE_MDFILE_AS_MAINPAGE = README.md
+
+RECURSIVE              = YES
+OUTPUT_DIRECTORY       = doc_output
+
+EXTRACT_ALL            = YES
+SORT_MEMBER_DOCS       = NO
+
+GENERATE_LATEX         = NO
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             = ROSIDL_TYPESUPPORT_C_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
```

Comment:
Where is this relative path coming from?

Reply:
It's not pretty, but it points to the same tag files directory used in the GENERATE_TAGFILE line below. `doxygen_tag_files` is added to a temporary worksource in the ros2 distro doxygen [build script](https://github.com/ros2/docs.ros2.org/blob/doc_gen/build_docs.sh).

It also needs to agree with the path of other doxygen files, like [rcl's doxyfile](https://github.com/ros2/rcl/blob/master/rcl/Doxyfile).

Reply:
:zipper_mouth_face: That sounds like an ugly hack - but since it seems to be what is being done in other repos...